task/transcode.go

```diff
@@ -0,0 +1,237 @@
+package task
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/livepeer/joy4/av"
+	"github.com/livepeer/joy4/av/avutil"
+	"github.com/livepeer/joy4/format"
+	"github.com/livepeer/joy4/format/mp4"
+	"github.com/livepeer/joy4/format/ts"
+	"github.com/livepeer/livepeer-data/pkg/data"
+	"github.com/livepeer/stream-tester/model"
+	"github.com/livepeer/stream-tester/segmenter"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	segLen = 5 * time.Second
+)
+
+func init() {
+	format.RegisterAll()
+	rand.Seed(time.Now().UnixNano())
+}
+
+func TaskTranscode(tctx *TaskContext) (*data.TaskOutput, error) {
+	var (
+		ctx             = tctx.Context
+		inputPlaybackID = tctx.InputAsset.PlaybackID
+		lapi            = tctx.lapi
+	)
+
+	fullPath := videoFileName(inputPlaybackID)
+	fir, err := tctx.inputOS.ReadData(ctx, fullPath)
+	if err != nil {
+		panic(err)
+	}
+	fileInMem, err := io.ReadAll(fir.Body)
```

Comment:
Would it be possible/easy to buffer this file on disk instead of in memory? We are currently running the `task-runner` with a `4Gi` memory limit, and I think we should ideally not use that much, so it would be nice to lower the footprint as well. 

Are the outputs being buffered as well btw? Could they be "buffered" on disk instead?

Reply:
Yes, it's the major point for optimisation.
But I was not sure what is available in our k8s, and even if `os.TempDir` will return writeable dir. So I went with safest option for now.

Reply:
Alright makes sense! Thx for sharing your concerns